test/opt/test_eliminate_dead_const.cpp

```diff
@@ -0,0 +1,717 @@
+                "%dead_spec_int_a_vec = OpSpecConstantComposite %v2int %dead_spec_int_a %dead_spec_int_a",
+
+                "%dead_spec_int_b = OpSpecConstant %int 2",
+                "%dead_spec_int_b_vec = OpSpecConstantComposite %v2int %dead_spec_int_b %dead_spec_int_b",
+
+                "%dead_const_int_c = OpConstant %int 3",
+                "%dead_const_int_c_vec = OpConstantComposite %v2int %dead_const_int_c %dead_const_int_c",
+
+                // Add
+                "%add_a_b = OpSpecConstantOp %int IAdd %dead_spec_int_a %dead_spec_int_b",
+                "%add_a_b_vec = OpSpecConstantOp %v2int IAdd %dead_spec_int_a_vec %dead_spec_int_b_vec",
+
+                // Sub
+                "%sub_a_b = OpSpecConstantOp %int ISub %dead_spec_int_a %dead_spec_int_b",
+                "%sub_a_b_vec = OpSpecConstantOp %v2int ISub %dead_spec_int_a_vec %dead_spec_int_b_vec",
+
+                // Mul
+                "%mul_a_b = OpSpecConstantOp %int IMul %dead_spec_int_a %dead_spec_int_b",
+                "%mul_a_b_vec = OpSpecConstantOp %v2int IMul %dead_spec_int_a_vec %dead_spec_int_b_vec",
+
+                // Div
+                "%div_a_b = OpSpecConstantOp %int SDiv %dead_spec_int_a %dead_spec_int_b",
+                "%div_a_b_vec = OpSpecConstantOp %v2int SDiv %dead_spec_int_a_vec %dead_spec_int_b_vec",
+
+                // Bitwise Xor
+                "%xor_a_b = OpSpecConstantOp %int BitwiseXor %dead_spec_int_a %dead_spec_int_b",
+                "%xor_a_b_vec = OpSpecConstantOp %v2int BitwiseXor %dead_spec_int_a_vec %dead_spec_int_b_vec",
+
+                // Scalar Comparison
+                "%less_a_b = OpSpecConstantOp %int SLessThan %dead_spec_int_a %dead_spec_int_b",
+            },
+        },
+
+        // Swizzles.
+        {
+            /* .used_consts = */
+            {
+                "%used_int = OpConstant %int 3",
+            },
+            /*  .main_insts = */
+            {
+                "%int_var = OpVariable %_pf_int Function",
+                "OpStore %int_var %used_int",
+            },
+            /* .dead_consts = */
+            {
+                "%dead_int = OpConstant %int 3",
+
+                "%dead_spec_int_a = OpSpecConstant %int 1",
+                "%vec_a = OpSpecConstantComposite %v4int %dead_spec_int_a %dead_spec_int_a %dead_int %dead_int",
+
+                "%dead_spec_int_b = OpSpecConstant %int 2",
+                "%vec_b = OpSpecConstantComposite %v4int %dead_spec_int_b %dead_spec_int_b %used_int %used_int",
+
+                // Extract scalar
+                "%a_x = OpSpecConstantOp %int CompositeExtract %vec_a 0",
+                "%b_x = OpSpecConstantOp %int CompositeExtract %vec_b 0",
+
+                // Extract vector
+                "%a_xy = OpSpecConstantOp %v2int VectorShuffle %vec_a %vec_a 1 0",
+                "%b_xy = OpSpecConstantOp %v2int VectorShuffle %vec_b %vec_b 1 0",
+            },
+        },
+        // clang-format on
+    })));
+
```

Comment:
It would be nice to have another spotcheck test case that has very (well, comparatively) complex references of used and dead constants to increase the confidence level. :)